compiler/luci/pass/src/FuseMulWithFullyConnectedPass.cpp

```diff
@@ -118,7 +118,39 @@ bool fuse_mul_with_fc(luci::CircleMul *mul)
   luci::CircleFullyConnected *fc = nullptr;
   luci::CircleConst *multiplication = nullptr;
   RETURN_FALSE_UNLESS(luci::fill(&fc, &multiplication).with_commutative_args_of(mul));
-  // Make sure that FullyConnected has only one successor:
+  /**
+   *  Make sure that FullyConnected has only one successor.
+   *
+   *  If the FullyConnected output is connected to more nodes,
+   *  this pass will replace node with new fused FullyConnected.
+   *  Thus pass success will only introduce extra FullyConnected
+   *  without reducing overall number of nodes.
+   *  Which tends to increase model's size and degrades model's performance.
+   *  Thus one successor is required to benefit from this pass.
+   *
+   *  Example graph that illustrates the described scenario:
+   *
+   *  BEFORE
+   *                |
+   *      [CircleFullyConnected]
+   *                |
+   *        +-------+----------------+
+   *        |                        |
+   *        |                        |
+   *  [Other Node]              [CircleMul]
+   *        |                        |
+   *
+   *  AFTER
+   *                |
+   *      [CircleFullyConnected]
+   *                |
+   *        +-------+-----------------------+
+   *        |                               |
+   *        |                               |
+   *  [Other Node]       [New CircleFullyConnected Fused with Mul]
```

Comment:
this looks incorrect

Reply:
Yes, spotted this after pushing... will change in a minute.